strawberry/extensions/query_depth_limiter.py

```diff
@@ -220,18 +236,62 @@ def is_ignored(node: FieldNode, ignore: Optional[List[IgnoreType]] = None) -> bo
     if ignore is None:
         return False
 
+    field_name = node.name.value
     for rule in ignore:
-        field_name = node.name.value
         if isinstance(rule, str):
             if field_name == rule:
                 return True
         elif isinstance(rule, re.Pattern):
             if rule.match(field_name):
                 return True
+        elif isinstance(rule, FieldAttributesRule):
+            return validate_field_attributes(node, rule)
         elif callable(rule):
             if rule(field_name):
                 return True
         else:
             raise TypeError(f"Invalid ignore option: {rule}")
 
     return False
+
+
+def validate_field_name(node: FieldNode, rule: FieldNameRuleType) -> bool:
+    field_name = node.name.value
+    if isinstance(rule, str):
+        if field_name == rule:
+            return True
+    elif isinstance(rule, re.Pattern):
+        if rule.match(field_name):
+            return True
+    elif callable(rule):
+        if rule(field_name):
+            return True
+    else:
+        raise ValueError(f"Invalid ignore option: {rule}")
+
```

Comment:
do you think we can add coverage for these lines? :)

Reply:
With the codecov report, I'll go through the additions to ensure it's all covered!